### PR TITLE
feat: use git init.defaultBranch if set

### PR DESCRIPTION
**Pull request by tvlieg**
My first line of fish shell script, so please let me know if did anything off the hook!

The main goal of this PR is to use the return value for `git config --get init.defaultBranch`, if set, for function `__git.default_branch.fish`.

### CODE REVIEW

**Comment by jhillyerd** [comment]
Hi, thanks for sending this, I wasn't aware of this setting.

Unfortunately, this change would break folks, as the global git setting will be returned if the local repository does not have it configured.  As an example, this repo:

```
❯ omf cd git
git on master
❯ git config --get init.defaultBranch
main
```

**Comment by tvlieg** [comment]
That's a fair point; I've pushed a new commit that first checks for the existence of the configured defaultBranch, and only returns it if the current repo has that branch. Would that work?

**Comment by tvlieg** [comment]
Btw, I'm running into the issue in which I have a repo with both a master, but also a develop branch, in which I would like the default branch to be develop.

File: functions/__git.default_branch.fish [hunk 1]
```diff
@@ -1,7 +1,10 @@
 function __git.default_branch -d "Fallback to main when master branch is not found"
   # Silently fail when not in git directory
   command git rev-parse --git-dir &>/dev/null; or return
-  if command git show-ref -q --verify refs/heads/master
+  set defaultBranch (command git config --get init.defaultBranch)
```
Comment (jhillyerd) [inline functions/__git.default_branch.fish:4]: I think we'll want to use `-l` (local) on this set, given the user might have a wider scoped defaultBranch variable defined, which we would then mangle.

> 
> • Variables may be explicitly set to universal, global or local. Variables with the  same
>   name in different scopes will not be changed.
> 
> • If  a  variable  is not explicitly set to be either universal, global or local, but has
>   been previously defined, the previous variable scope is used.

**Review by jhillyerd — CHANGES_REQUESTED** [review]

File: functions/__git.default_branch.fish (hunk 1 above)
Comment (tvlieg) [reply functions/__git.default_branch.fish]: pushed a new revision :)

**Comment by jhillyerd** [comment]
Thanks!

**Comment by tvlieg** [comment]
And thank you as well for thinking along :)

**Pull request merged**